service/worker/scanner/build_ids/scavenger.go

```diff
@@ -67,8 +67,9 @@ var (
 
 type (
 	BuildIdScavangerInput struct {
-		NamespaceListPageSize int
-		TaskQueueListPageSize int
+		NamespaceListPageSize  int
+		TaskQueueListPageSize  int
+		OverrideTimeForTesting *time.Time
```

Comment:
Can we have a flag to ignore retention instead?
It may be useful to trigger this workflow with that setting if a user accidentally creates many unused build IDs.